### DISCUSSION TITLE
Logging - Trim newlines and white chars

### DISCUSSION THREAD

**Discussion by Karql** [jertel/elastalert2#650] [Ideas]
Hi!

Some logs like: 

![image](https://user-images.githubusercontent.com/173098/148635996-c45c7634-c0a7-449a-a780-b9d25bfc9239.png)

add new lines and white characters to output.

This behavior makes logs parsing more difficult.

I've seen logging section in configuration but as I've searched it is not possible to strip log message using formatter.

Maybe you could change these multi-line logs?

Best regards,
Mateusz

**Comment by nsano-rururu** [comment]
If you know where to fix it, please pull request. There is no rule that you should not make a pull request.

**Reply by nsano-rururu** [reply]
By the way, does this mean you want it fixed for elastalert2-server?

**Reply by nsano-rururu** [reply]
What is a formatter?

**Reply by nsano-rururu** [reply]
Are there any other similar parts? .. Is it just the point you pointed out?

**Reply by nsano-rururu** [reply]
I don't think it's a question that takes time to answer. Are you ignoring me?

**Reply by nsano-rururu** [reply]
I don't know what happened in the past, but I think it was implemented in that way to make it stand out in debug mode.

**Reply by nsano-rururu** [reply]
In the first place, it seems that you are requesting to change it because you are in trouble when you capture the log in elasticsearch and display it, but is it different?

**Reply by nsano-rururu** [reply]
If there is a counterargument, please refute it. I feel sick

**Comment by Karql** [comment]
> If you know where to fix it, please pull request. There is no rule that you should not make a pull request.

I'd like to start discussion first.
Btw. I am not a python guy so I would prefer not to make changes.

> By the way, does this mean you want it fixed for elastalert2-server?

Yes

> What is a formatter?

https://github.com/jertel/elastalert2/blob/master/examples/config.yaml.example#L91

![image](https://user-images.githubusercontent.com/173098/148654697-e6790439-b307-4023-82aa-3f3f10cfafcd.png)

https://stackoverflow.com/questions/17931426/strip-newline-and-white-spaces-from-python-logger-messages

> Are there any other similar parts? .. Is it just the point you pointed out?

This is the first place I've noticed. I have not reviewed all source code.

> I don't think it's a question that takes time to answer. Are you ignoring me?

I was away from the computer.

> In the first place, it seems that you are requesting to change it because you are in trouble when you capture the log in elasticsearch and display it, but is it different?

When log is splited into multiple lines it is difficult to for expample caputre it by fluentd.

**Reply by nsano-rururu** [reply]
I don't want to do it because I don't know python. Do not be silly. Then I don't know python either, but I added the function of elastalert2 and fixed the bug.

**Reply by nsano-rururu** [reply]
Does it mean that the contents of the docker log are imported into elasticsearch with fluentd with elastalert2-server? Explain it properly.

**Reply by nsano-rururu** [reply]
Why do I have to spend my private time fixing for elastalert2-server.

**Reply by nsano-rururu** [reply]
You are the maintainer of elastalert2-server, aren't you? You can write a program, right?

**Reply by nsano-rururu** [reply]
If there are multiple lines, if an exception occurs, the stack trace will be multiple lines. What can I do with fluentd?

**Reply by nsano-rururu** [reply]
elastalert2-server is a third party, not an integrated official tool. Can you understand? I'm the maintainer of praeco and elastalert-server, but this is just a third party tool and I haven't made any requests to elastalert2 for the convenience of the tool. Also, I have the commit right of elastalert2, but basically I try to make a pull request.

**Reply by nsano-rururu** [reply]
Is it because the user asked me to change it, rather than because of elastalert2-server? .. I feel something is wrong

**Reply by nsano-rururu** [reply]
It feels strange to write another person's request on your behalf rather than what you want it to do.

**Reply by nsano-rururu** [reply]
If you want to change it, please make a pull request. The story ends.

**Reply by nsano-rururu** [reply]
Check it out. You don't know much about fluentd

Fluentdで複数行のログを読む
https://kazuhira-r.hatenablog.com/entry/2021/01/28/000628
Fluentdのin_tailプラグインで複数行のログをよむ方法
https://kazegusuri.hateblo.jp/entry/2014/06/28/201239
複数行のログをキーワード監視＆通知することで運用を少し改善しよう
https://qiita.com/uzresk/items/935baa47f83921158276
Multi format parser plugin for Fluentd
https://github.com/repeatedly/fluent-plugin-multi-format-parser
multiline
https://docs.fluentd.org/parser/multiline

**Reply by nsano-rururu** [reply]
Please see the project guidelines properly.
https://github.com/jertel/elastalert2/discussions/11

**Reply by nsano-rururu** [reply]
Ask questions about fluentd in the fluentd community.
https://github.com/fluent/fluentd

**Comment by Karql** [comment]
fluentd is just an example.

It always a good practice to keep logs in single lines or add possibility to configure it in that way e.g. `structured logging`

So this is not:
> fixing for elastalert2-server.

but idea how to improve this project...

> Please see the project guidelines properly.

IMO you should do it first.

**Reply by nsano-rururu** [reply]
@Karql 

Pull requests are welcome. The story should be over.

**Reply by nsano-rururu** [reply]
@Karql

If you want to change it, do it yourself.

**Reply by nsano-rururu** [reply]
elastalert2-server is not an official tool. You are the one who thinks about it.

>fixing for elastalert2-server.
>
>but idea how to improve this project...

**Reply by nsano-rururu** [reply]
Do you understand OSS? In the first place, there is no obligation to fix bugs or respond to requests. I was very disappointed.

**Reply by nsano-rururu** [reply]
The `johnsusek/elastalert-server` and `johnsusek/praeco` I maintain are also recognized as unofficial tools.

**Reply by nsano-rururu** [reply]
It is my policy to be strict with people and strict with myself.
